src/MarginTrading.Backend.Services/Caches/OvernightSwapCache.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using MarginTrading.Backend.Core;
+using MoreLinq;
+
+namespace MarginTrading.Backend.Services.Caches
+{
+	public class OvernightSwapCache : IOvernightSwapCache
+	{
+		private Dictionary<string, OvernightSwapCalculation> _cache;
+
+		private static readonly object LockObj = new object();
+
+		public OvernightSwapCache()
+		{
+			ClearAll();
+		}
+        
+		public OvernightSwapCalculation Get(string key)
+		{
+			lock (LockObj)
+			{
+				return _cache.TryGetValue(key, out var value)
+					? value
+					: null;
+			}
+		}
+
+		public bool TryGet(string key, out OvernightSwapCalculation item)
+		{
+			lock (LockObj)
+			{
+				return _cache.TryGetValue(key, out item);
+			}
+		}
+
+		public IReadOnlyList<OvernightSwapCalculation> GetAll()
+		{
+			lock(LockObj)
+			{
+				return _cache.Values.ToList();
+			}
+		}
+
+		public bool TryAdd(OvernightSwapCalculation item)
+		{
+			if (item == null || _cache.ContainsKey(item.Key))
+				return false;
+            
+			lock (LockObj)
+			{
+				_cache[item.Key] = item;
+				return true;
+			}
+		}
+
+		public bool AddOrReplace(OvernightSwapCalculation item)
+		{
+			if (item == null)
+				return false;
+			
+			lock (LockObj)
+			{
+				_cache.Remove(item.Key);
+				_cache[item.Key] = item;
+			}
+
+			return true;
+		}
+
+		public void SetAll(IEnumerable<OvernightSwapCalculation> items)
```

Comment:
May be it is better to make it private? It is used only in Initialize

Reply:
+